`TestClusterTables/ForClusterServerInfo` fails on apple M1

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

`make test`
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

all cases pass
### 3. What did you see instead (Required)

```
--- FAIL: TestClusterTables (2.11s)
    --- FAIL: TestClusterTables/ForClusterServerInfo (1.53s)
        cluster_tables_serial_test.go:154:
                Error Trace:    cluster_tables_serial_test.go:154
                Error:          Not equal:
                                expected: set.StringSet{"cpu":struct {}{}, "disk":struct {}{}, "memory":struct {}{}, "net":struct {}{}}
                                actual  : set.StringSet{"disk":struct {}{}, "memory":struct {}{}, "net":struct {}{}}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,4 +1,2 @@
                                -(set.StringSet) (len=4) {
                                - (string) (len=3) "cpu": (struct {}) {
                                - },
                                +(set.StringSet) (len=3) {
                                  (string) (len=4) "disk": (struct {}) {
                Test:           TestClusterTables/ForClusterServerInfo
                Messages:       sql: select * from information_schema.CLUSTER_HARDWARE;
```

### 4. What is your TiDB version? (Required)

master branch
golang 1.17.2
apple M1

<!-- Paste the output of SELECT tidb_version() -->

I think the root cause is https://github.com/pingcap/sysutil/blob/e5c5906ed1e736d38565158bc6da9e3f2ee07034/hardware_info.go#L18. This call will fail on apple M1 due to https://github.com/shirou/gopsutil/issues/1000.

--- a/infoschema/cluster_tables_serial_test.go
+++ b/infoschema/cluster_tables_serial_test.go
@@ -97,11 +97,11 @@ func SubTestForClusterServerInfo(s *clusterTablesSuite) func(*testing.T) {
 		defer func() { require.NoError(t, failpoint.Disable(fpName)) }()
 
 		cases := []struct {
-			sql      string
-			types    set.StringSet
-			addrs    set.StringSet
-			names    set.StringSet
-			skipOnOS string
+			sql        string
+			types      set.StringSet
+			addrs      set.StringSet
+			names      set.StringSet
+			skipOnDist set.StringSet
 		}{
 			{
 				sql:   "select * from information_schema.CLUSTER_LOAD;",
@@ -115,7 +115,8 @@ func SubTestForClusterServerInfo(s *clusterTablesSuite) func(*testing.T) {
 				addrs: set.NewStringSet(s.listenAddr),
 				names: set.NewStringSet("cpu", "memory", "net", "disk"),
 				// The sysutil package will filter out all disk don't have /dev prefix.
-				skipOnOS: "windows",
+				// gopsutil cpu.Info will fail on mac M1
+				skipOnDist: set.NewStringSet("windows", "darwin/arm64"),
 			},
 			{
 				sql:   "select * from information_schema.CLUSTER_SYSTEMINFO;",
@@ -126,12 +127,12 @@ func SubTestForClusterServerInfo(s *clusterTablesSuite) func(*testing.T) {
 				// Because the underlying implementation use `sysctl` command to get the result
 				// and there is no such command on windows.
 				// https://github.com/pingcap/sysutil/blob/2bfa6dc40bcd4c103bf684fba528ae4279c7ec9f/system_info.go#L50
-				skipOnOS: "windows",
+				skipOnDist: set.NewStringSet("windows"),
 			},
 		}
 
 		for _, cas := range cases {
-			if cas.skipOnOS == runtime.GOOS {
+			if cas.skipOnDist.Exist(runtime.GOOS+"/"+runtime.GOARCH) || cas.skipOnDist.Exist(runtime.GOOS) {
 				continue
 			}
 

